commands/api-gateway/gateway/delete.go

```diff
@@ -0,0 +1,84 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/api-gateway/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/confirm"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/functional"
+	"github.com/ionos-cloud/sdk-go-bundle/products/apigateway/v2"
+	"github.com/spf13/viper"
+)
+
+func ApiGatewayDeleteCmd() *core.Command {
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Namespace: "apigateway",
+		Resource:  "gateway",
+		Verb:      "delete",
+		Aliases:   []string{"del", "d"},
+		ShortDesc: "Delete a gateway",
+		Example:   "ionosctl apigateway gateway delete --gateway-id ID",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			return core.CheckRequiredFlagsSets(c.Command, c.NS, []string{constants.ArgAll}, []string{constants.FlagGatewayID})
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			if all := viper.GetBool(core.GetFlagName(c.NS, constants.ArgAll)); all {
+				return deleteAll(c)
+			}
+
+			apigatewayId := viper.GetString(core.GetFlagName(c.NS, constants.FlagGatewayID))
+			z, _, err := client.Must().Apigateway.APIGatewaysApi.ApigatewaysFindById(context.Background(), apigatewayId).Execute()
+			if err != nil {
+				return fmt.Errorf("failed getting gateway by id %s: %w", apigatewayId, err)
+			}
+			yes := confirm.FAsk(c.Command.Command.InOrStdin(), fmt.Sprintf("Are you sure you want to delete gateway %s ", z.Properties.Name),
+				viper.GetBool(constants.ArgForce))
+			if !yes {
+				return fmt.Errorf(confirm.UserDenied)
+			}
+
+			_, err = client.Must().Apigateway.APIGatewaysApi.ApigatewaysDelete(context.Background(), apigatewayId).Execute()
+			if err != nil {
```

Comment:
Empty if . I get that the error is returned below, but either add the `if` or return the err, not both